### PR TITLE
[HR] review Employees Filters

### PR DESCRIPTION
**Pull request by dka-odoo**
Task : https://www.odoo.com/web#id=10497&view_type=form&model=project.task&menu_id=3942&action=327
Pad  : https://pad.odoo.com/p/r.Om4jUh4rwMsAPFrO

### CODE REVIEW

**Comment by antonylesuisse** [comment]
Dont override search.

in open_view_employee_list action set an evaluted context to set

user_departement_id

Then in search view use filter ['departement_id',=, context.get('user_departement_id')]

**Comment by dka-odoo** [comment]
from where we set context in 'open_view_employee_list' action?  as i am not able to find the current user's employee department in xml,  @antonylesuisse   can you give your suggestion please? thanks

**Comment by antonylesuisse** [comment]
addons/hr/hr_view.xml:220 you need to change the action by a server action that return an ir.action.act_window with a context containing user_departement_id.

File: addons/hr/hr_view.xml [hunk 1]
```diff
@@ -133,6 +133,7 @@
                     <field name="category_ids" groups="base.group_hr_user"/>
                     <field name="job_id" />
                     <filter string="New Mail" name="message_unread" domain="[('message_unread','=',True)]"/>
+                    <filter string="My Department" domain="[('department_id', '=', context.get('user_department_id', False))]"/>
```
Comment (rim-odoo) [inline addons/hr/hr_view.xml]: This filter is not working when I try to use it...

File: addons/hr/hr.py [hunk 2]
```diff
@@ -336,6 +336,15 @@ def action_unfollow(self, cr, uid, ids, context=None):
             that receive the context without the wrapper. """
         return self.message_unsubscribe_users(cr, uid, ids, context=context)
 
+    def action_open_employee_kanban(self, cr, uid, context=None):
+        emp_id = self.search(cr, uid, [('user_id', '=', uid)], limit=1, context=context)
+        department_id = emp_id and self.browse(cr, uid, emp_id, context=context).department_id or False
+
+        action = self.pool['ir.model.data'].xmlid_to_object(cr, uid, 'hr.open_view_employee_list_my', context=context)
+        result = action.read()[0]
+        result['context'] = {'user_department_id': department_id and department_id.id}
```
Comment (rim-odoo) [inline addons/hr/hr.py]: What if action had a context? Is it lost?

**Pull request closed without merging**